app/src/androidTest/java/org/mozilla/fenix/ui/SearchTest.kt

```diff
@@ -298,4 +298,52 @@ class SearchTest {
             verifyRecentlyVisitedSearchGroupDisplayed(false, "test search", 1)
         }
     }
+
+    // Default search code for Google-US
+    @Test
+    fun defaultSearchCodeGoogleUS() {
+        homeScreen {
+        }.openSearch {
+        }.submitQuery("firefox") {
+            verifyUrl("client=firefox-b-m")
+        }
+    }
+
+    // Default search code for Bing-US
+    @Test
+    fun defaultSearchCodeBingUS() {
+        homeScreen {
+        }.openThreeDotMenu {
+        }.openSettings {
+        }.openSearchSubMenu {
+            changeDefaultSearchEngine("Bing")
+        }
+
+        exitMenu()
+
+        homeScreen {
+        }.openSearch {
+        }.submitQuery("firefox") {
+            verifyUrl("firefox&pc=MOZB&form=MOZMBA")
+        }
+    }
+
+    // Default search code for DuckDuckGo-US
+    @Test
+    fun defaultSearchCodeDuckDuckGoUS() {
+        homeScreen {
+        }.openThreeDotMenu {
+        }.openSettings {
+        }.openSearchSubMenu {
+            changeDefaultSearchEngine("DuckDuckGo")
+        }
+
+        exitMenu()
+
+        homeScreen {
+        }.openSearch {
+        }.submitQuery("firefox") {
+            verifyUrl("t=fpas")
+        }
+    }
```

Comment:
I would create a reusable variable for the query string and possibly map the codes to their search engine names in another object.

Reply:
Done !